src/core.ts

```diff
@@ -25,12 +25,18 @@ export interface CollationSpec {
   readonly backwards?: boolean // unsupported
 }
 
+/**
+ * Custom function to hash values to improve faster comparaisons
+ */
+export type CustomHashFunction =  (v: string) => string | number
```

Comment:
**recommendation:** could we simplify the type name to `HashFunction`

Reply:
done, as you said